python/ql/lib/semmle/python/Patterns.qll

```diff
@@ -0,0 +1,123 @@
+/**
+ * Wrapping generated AST classes: `Pattern_` and subclasses.
+ */
+
+import python
+
+/** A pattern in a match statement */
+class Pattern extends Pattern_, AstNode {
+  /** Gets the scope of this pattern */
+  override Scope getScope() {
+    // TODO: Should it be defined as
+    //   py_scopes(this, result)
+    // instead?
+    result = this.getCase().getScope()
```

Comment:
I think reducing it to the scope of the surrounding `case` makes sense. Does using `py_scopes` actually make any observable difference?

Reply:
I think that using `py_scopes` requires some work which is why I did it like this. The `TODO` was to remind me to ask the reviewers if that would be fine :-)